lib/python/picongpu/extra/plugins/data/radiation.py

```diff
@@ -125,3 +126,40 @@ def get_vector_n(self):
         n_vec[:, 1] = n_y * n_y_unitSI
         n_vec[:, 2] = n_z * n_z_unit_SI
         return n_vec
+
+    def get_distributedAmplitude(self):
+        """"
+        Return a 4D complex array of all distributed amplitudes
+        Array dimensions:
+        0: index GPU/worker
+        1: direction
+        2: frequency
+        3: polarization [x, y, z]
+        returns None if no distributed amplitudes are available
+        """
+        try:
+            h_distAmp = self.iteration.meshes["Amplitude_distributed"]
+        except IndexError:
+            warnings.warn("Warning: no distributed amplitude available.")
+            return None
+
+        # get shape of distributed amplitude
+        shapeDistAmp = h_distAmp["x"].shape
+        # add 3 vector components
+        shapeDistAmp.append(3)
+
+        # prepare data array
+        distAmp = np.empty(shape=shapeDistAmp, dtype=np.complex128)
```

Comment:
Do you really want to create an empty array here, which has garbage values inside? Wouldn't zeroes be better to spot errors ? (Some day, maybe, the size of the y and z components do not equal the size of the x component. Although the assignment later should then throw an error)

I don't have an opinion on this, actually.

Reply:
Since there is no way of ever seeing the empty array since it is filled in the code lines below, I favor having only the empty allocation and no extra setting to zero. However, the array won't be very big, so this optimization is only marginal.

Reply:
I strongly agree with @steindev: Either you say that this is a premature optimisation (and we all know about how it relates to the tree structure of all evil) and use `np.zeros` or you say that it's necessary, then please make sure that what you say is true by extracting the allocation and initialisation into a separate function. That way you can make it part of the contract of the function that the returned array is always correctly initialised.

Reply:
@steindev and I had just a brief *offline* discussion. Initializing with NaNs will be better because it clearly indicates something going wrong.